datafusion/core/src/physical_plan/sorts/sort_preserving_merge.rs

```diff
@@ -388,22 +389,23 @@ impl SortPreservingMergeStream {
                 }
                 Some(Ok(batch)) => {
                     if batch.num_rows() > 0 {
-                        let cursor = match SortKeyCursor::new(
+                        let cols = self
+                            .column_expressions
+                            .iter()
+                            .map(|expr| {
+                                Ok(expr.evaluate(&batch)?.into_array(batch.num_rows()))
+                            })
+                            .collect::<Result<Vec<_>>>()?;
+
+                        let rows = self.row_converter.convert(&cols);
+
+                        let cursor = SortKeyCursor::new(
```

Comment:
We need to track the total memory used by all cursors since the cursor now holds `Rows`. We could do this as follow-ups but note here as it came to me.

Reply:
I agree that memory usage is a potential concern (as we are effectively copying data into the `Rows` format. 

A follow on PR would be good I think. I filed https://github.com/apache/arrow-datafusion/issues/3609